app/services/create_post.rb

```diff
@@ -1,10 +1,16 @@
 class CreatePost
+  class InvalidPostType < StandardError; end
+
   def self.call(group, membership, post_type_id)
-    if post_type_id == PostType::LEADER_POST_ID
-      remove_leader(group)
-      remove_past_leader_post(membership)
+    raise InvalidPostType unless group.has_post_type?(post_type_id)
```

Comment:
One question, what happens with the [default posts](https://github.com/kir-dev/pek-next/blob/master/app/models/post_type.rb#L22L27) that has no group associated with?

Reply:
I haven't thought about this, but I checked it. All group has a post with DEFAULT_POST_ID, so this PR doesn't prevent the users from joining a group.

Reply:
The `has_post_type?` method includes the default posts, they are gathered in the [post_types](https://github.com/kir-dev/pek-next/blob/039574450c0a391fb44976820db7a4ea11f7b62f/app/models/group.rb#L100) method.